### ISSUE TITLE
Sik-Ho Tang | Review -- Motion Masks: Learning Features by Watching Objects Move.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#127]
Sik-Ho Tang. [Review — Motion Masks: Learning Features by Watching Objects Move](https://sh-tsang.medium.com/review-motion-masks-learning-features-by-watching-objects-move-548e7d955c2e).

**Comment by NorbertZheng** [comment]
## Overview
Outperforms [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff), [Split-Brain Auto](https://sh-tsang.medium.com/review-split-brain-autoencoders-unsupervised-learning-by-cross-channel-prediction-cebeba644a3c), [Jigsaw](https://sh-tsang.medium.com/review-unsupervised-learning-of-visual-representations-by-solving-jigsaw-puzzles-50b14d755004), [Context Encoders](https://sh-tsang.medium.com/review-context-encoders-feature-learning-by-inpainting-bd181e48997), [BiGAN](https://sh-tsang.medium.com/review-bigan-adversarial-feature-learning-gan-535eb76be2ca), [Wang ICCV’15](https://sh-tsang.medium.com/review-unsupervised-learning-of-visual-representations-using-videos-abee72149f77), etc.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/5284c30a-2d8a-4f89-88eb-9a44e05fe62f)
Motion helps us to correctly group pixels that move together (bottom left) and identify this group as a single object (bottom right).

Learning Features by Watching Objects Move Motion Masks, by Facebook AI Research (FAIR), and University of California. 2017 CVPR, Over 400 Citations.
Unsupervised Learning, Representation Learning, Object Detection, Image Classification, Semantic Segmentation, Video Classification.
 - Motion plays a key role in the development of the human visual system.
 - In this paper, an unsupervised motion-based segmentation on videos is proposed to obtain segments, which is treated as ‘pseudo ground truth’ to train a convolutional network to segment objects from a single frame.
 - When used for transfer learning on object detection, the proposed representation significantly outperforms previous unsupervised approaches.

**Comment by NorbertZheng** [comment]
Introduce time inductive bias (specific to vision signals!!!)

**Comment by NorbertZheng** [comment]
## Approach Overview
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/951b28c0-c5ab-453c-bd29-b54817375a01)
Approach Overview.

Yahoo Flickr Creative Commons 100 million (YFCC100m) [43] dataset is used.
 - **Motion cues are used to segment objects in videos without any supervision.**

Then, a ConvNet is trained to **predict these segmentations** from static frames, i.e. without any motion cues.

The learned representation is transferred to other recognition tasks.

**Comment by NorbertZheng** [comment]
Pseudo-Segmentation Task!!! Automatic segmentation annotation by motion (x

**Comment by NorbertZheng** [comment]
## Unsupervised Motion Segmentation
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/cb49dba8-ca26-4e1d-9317-a5368e17041f)
Left: a video frame, Right: the output of uNLC.

Algorithm:
 - The NLC approach from Faktor and Irani [12] is unsupervised with respect to video segmentation, it utilizes an edge detector that was trained on labeled edge images.
 - uNLC is proposed where a nearest neighbor graph is computed over the superpixels in the video using location and appearance (color histograms and HOG [6]) as features, so that it is totally unsupervised.

Frames are discarded:
 - frames with too many (>80%) or too few (<10%) pixels marked as foreground;
 - frames with too many pixels (>10%) within 5% of the frame border that are marked as foreground.

Dataset:
 - uNLC is run on videos from YFCC100m [43], which contains about 700,000 videos. It ends up with 205,000 videos after pruning.
 - 5–10 frames per shot from each video are sampled to create the dataset of 1.6M (i.e. 1'600'000) images, so we have slightly more frames than images in ImageNet.

**Comment by NorbertZheng** [comment]
## Training a ConvNet to Segment Objects
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/471a091c-8cd4-45a1-93b4-f82ba69be780)
Left: A video frame, Middle: the output of uNLC that we use to train the ConvNet, and Right: the output of the ConvNet.

Pretext Task:
 - The ConvNet is trained to segment the object, i.e., assign each pixel a label of 1 if it lies on the object and 0 otherwise.
 - An object is sampled from an image and a box is cropped around the ground truth segment to make sure
    - **only one object exists**,
    - **the box is jittered in position and scale.**

Model:
 - [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160) is used. It takes as input a $w\times w$ image and outputs an $s\times s$ mask. $w=227$, $s=56$.
 - The proposed network ends in a fully connected layer with $s^{2}$ outputs followed by an element-wise sigmoid. The resulting $s^{2}$ dimensional vector is reshaped into an $s\times s$ mask.
 - **The ground truth mask is downsampled to $s\times s$ and the cross entropy losses are sum over the $s^{2}$ locations to train the network.**

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/47c4cceb-f9bd-4c43-874e-c3552753082c)
Examples of segmentations produced by our ConvNet on held out images.

**Comment by NorbertZheng** [comment]
## Experimental Results
### Does training for segmentation yield good features?
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/6452a4f9-2970-4ee1-a85a-931d6f16d3bd)
The proposed representation (Supervised Masks) trained on manually-annotated segments from COCO (without class labels) compared to ImageNet pretraining and [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff) (unsupervised).

Object detection on PASCAL VOC 2007 using [Fast R-CNN](https://medium.com/coinmonks/review-fast-r-cnn-object-detection-a82e172e87ba) is evaluated.

Results:
 - The proposed supervised representation outperforms the unsupervised [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff) model across all scenarios by a large margin, which is to be expected.
 - Notably though, the proposed model maintains a fairly small gap with ImageNet pretraining.

Thus, **given high-quality segments, the proposed proposed method can learn a strong representation**, which validates authors’ hypothesis.

The model trained on [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff) degrades rapidly as more layers are frozen.
 - **This drop indicates that the higher layers of the model have become overly specific to the pretext task.**
 - The proposed approach retains good performance even when most of the ConvNet is frozen, indicating that it has indeed learned high-level semantics in the higher layers.

**Comment by NorbertZheng** [comment]
Freezing higher layers may lead to keeping pretext-task-specific embeddings. But this does matter!!! The more layers we can use, the higher-level semantics the network has extracted.

**Comment by NorbertZheng** [comment]
### Can the ConvNet learn from noisy masks (uNLC masks???)?
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/8daba982-791a-42b7-8687-d3d0e8bc72a9)
From left to right, the original mask, dilated and eroded masks (boundary errors), and a truncated mask (truncation can be on any side).

The ground truth masks are degraded to measure the impact of segmentation quality on the learned representation.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/6b74b84a-eaa0-4d56-a9d8-1fe5445117ce)
VOC object detection accuracy using our supervised ConvNet as noise is introduced in mask boundaries, the masks are truncated, or the amount of data is reduced.

 - **Surprisingly, the representation maintains quality even with large degradation.**

**Comment by NorbertZheng** [comment]
Information redundancy is too high!!!

**Comment by NorbertZheng** [comment]
### How much data do we need?
As shown at the right of the above figure,
 - **building a good representation requires significant amounts of training data.**

**Comment by NorbertZheng** [comment]
- Large amount of DATA!!!
 - High performance GPUs!!!
 - Strong Engineering Ability!!!

**Comment by NorbertZheng** [comment]
### SOTA Comparison
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/6b113c2d-d079-4259-9bfe-221a08a901f8)
Object detection AP (%) on PASCAL VOC 2012 using [Fast R-CNN](https://medium.com/coinmonks/review-fast-r-cnn-object-detection-a82e172e87ba) with various pretrained ConvNets.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/0f1a74bf-2636-4b19-aa5c-6aa0558a0224)
Results on object detection using [Fast R-CNN](https://medium.com/coinmonks/review-fast-r-cnn-object-detection-a82e172e87ba).

Model:
 - For supervised learning, the first is trained on ImageNet classification. The second is trained on manually-annotated segments (without class labels) from COCO.

Results:
 - The proposed representation learned from unsupervised motion segmentation performs on par or better than prior work on unsupervised learning across all scenarios.
 - The training data is from different domains (YFCC100m videos vs. ImageNet images). The two variants perform similarly: 33.4% mean AP when trained on YFCC with conv5 and below frozen compared to 33.2% for the ImageNet version. **This confirms that the different image sources do not explain the proposed gains.**

**Comment by NorbertZheng** [comment]
### Low-Shot Transfer
 - In this scenario it actually hurts to fine-tune the entire network, and the best setup is to leave some layers frozen.
 - But the proposed network obtains the best performance, outperforms [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff), [Split-Brain Auto](https://sh-tsang.medium.com/review-split-brain-autoencoders-unsupervised-learning-by-cross-channel-prediction-cebeba644a3c), [Jigsaw](https://sh-tsang.medium.com/review-unsupervised-learning-of-visual-representations-by-solving-jigsaw-puzzles-50b14d755004), Inpainting ([Context Encoders](https://sh-tsang.medium.com/review-context-encoders-feature-learning-by-inpainting-bd181e48997)), [BiGAN](https://sh-tsang.medium.com/review-bigan-adversarial-feature-learning-gan-535eb76be2ca), Tracking Video ([Wang ICCV’15](https://sh-tsang.medium.com/review-unsupervised-learning-of-visual-representations-using-videos-abee72149f77)), etc.

**Comment by NorbertZheng** [comment]
Freeze parameters, low learning rate when fine-tuning.

**Comment by NorbertZheng** [comment]
### Other Downstream Tasks
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/3cd553d3-e642-42a0-89c4-892b9b245de1)
Results on image (object) classification on VOC 2007, single-image action classification on Stanford 40 Actions, and semantic segmentation on VOC 2011.

When the ConvNet is progressively frozen, the proposed approach is a strong performer.
 - When all layers until conv5 are frozen, the proposed representation is better than other approaches on action classification and second only to [Colorization](https://sh-tsang.medium.com/review-colorful-image-colorization-self-supervised-learning-c2667640bd0a) [51] on image classification on VOC 2007 and semantic segmentation on VOC 2011.

**Comment by NorbertZheng** [comment]
Have learned high-level semantics!!!

**Comment by NorbertZheng** [comment]
## Reference
 - [2017 CVPR] [Motion Masks] [Learning Features by Watching Objects Move](https://arxiv.org/abs/1612.06370).
